### DISCUSSION TITLE
Is this package maintained?

### DISCUSSION THREAD

**Discussion by p-baum** [coddingtonbear/django-mailbox#259] [General]
Issues seem to be piling up.  No support for django 4.  Are we done here?

**Comment by coddingtonbear** [comment]
Not done, no, but dependent upon community maintainers like yourself.  This is an open-source project; so it's nobody's job to maintain this, and any maintenance that is completed is dependent on the motivation and free time of community members.  If you find value in this project, I suggest you take a moment to see how you might be able to help out.

**Reply by Tompickett21** [reply]
Could you just commit people's changes?

**Reply by coddingtonbear** [reply]
Unfortunately, that's the easy part!  The harder part is taking the time to review PRs and manually testing changes.  Are you up for helping out with that?

**Reply by p-baum** [reply]
It's your project. You have to accept some responsibility. Merging pull requests is one of them.

**Reply by coddingtonbear** [reply]
Unfortunately for you, that's not true.  You have to understand that I'm volunteer here.  I am not paid to contribute to this project, so the amount of time I am willing to spend on it is entirely dependent upon my personal motivations, and I have to say that comments like yours reduce my willingness to put in further effort if they have any effect at all.  I work on these kinds of projects for fun, not because I have an obligation to you or anybody else.

Given that I've already volunteered probably hundreds of hours over the years putting this project together and maintaining it, perhaps, if you still find value in this project, you can chip in some of _your_ time to help contribute by vetting changes and verifying PRs posted here.

**Reply by p-baum** [reply]
Like you I don't have time.  But I never took on any responsibility in the first place. Thanks for taking the time to make things clear.  I now have a good understanding of the maintenance state of this repository and can make the right decision as to whether I should use it or not.

**Reply by coddingtonbear** [reply]
This project is a gift that you and the rest of the community were given by me and other contributors.  That that gift isn't exactly what you want or that it has grown less-useful over time does not mean that I or other contributors have a responsibility to spend _more_ of our free time for your benefit.

That isn't to say that I'm not willing to help out, though -- it's just that the maximum responsibility I'm able to take on at this point is merging and performing releases once PRs have been properly vetted.  The merging and releasing of things is simple and doesn't take much time at all; the actual vetting is not and does.

**Reply by p-baum** [reply]
Adding your position as you have described it here in a more prominent position may avoid future conversations like this.

**Reply by coddingtonbear** [reply]
I understand that this is inconvenient for you, but this is the standard reality of all open-source projects -- someone giving one gift of their time does not obligate them to give another. 

Honestly, I've contributed to quite a lot of projects over the last decade or so, in case you haven't noticed, and I believe this is the only occasion in which I've had to explain to somebody that they are not entitled to deciding what I do with my time.  Instead, what I'd recommend for avoiding future conversations like this, is reading the licenses of the software you're considering more carefully: https://github.com/coddingtonbear/django-mailbox/blob/master/LICENSE#L15 .

**Reply by darkpixel** [reply]
@coddingtonbear What's the process for getting a commit bit so I can take some of the load off?

**Reply by coddingtonbear** [reply]
Just to be clear, @darkpixel, because from above it seems like folks think the commit & merge is the thing I'm struggling to find time for: if merging and releasing were all that were needed, all of these PRs would've been closed years ago.  The thing that's really needed on this project (and most open-source projects, really) is for folks to test and review PRs.  Unfortunately, reviewing and testing PRs is a much more time-consuming process than just merging and performing a release.

To be concrete; what I need before I can merge any PR is:

- Folks to have reviewed the PR to see if they can find any code quality concerns, logical problems, bugs, missing tests, etc.
- Folks to have manually tested the PR and noted on the PR what sort of testing they performed.

If you can help out with those sorts of things over time, I'm not at all opposed to giving somebody like you the commit bit.
